src/components/forms/DomainSettings/KeywordRow.tsx

```diff
@@ -0,0 +1,228 @@
+import { PlusCircleFilled } from '@ant-design/icons';
+import { Pill } from '@src/components/data-display/Pill';
+import ConfirmTransactionModal from '@src/components/modals/ConfirmTransactionModal/ConfirmTransactionModal';
+import { ANT_INTERACTION_TYPES, ContractInteraction } from '@src/types';
+import eventEmitter from '@src/utils/events';
+import { Skeleton } from 'antd';
+import { Check, X } from 'lucide-react';
+import { useEffect, useRef, useState } from 'react';
+
+import DomainSettingsRow from './DomainSettingsRow';
+
+function AddKeywordInput({ addCb }: { addCb: (word: string) => void }) {
+  const [keyword, setKeyword] = useState<string>('');
+  const [edit, setEdit] = useState<boolean>(false);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  // Focus the input whenever edit mode is enabled
+  useEffect(() => {
+    if (edit) {
+      inputRef.current?.focus();
+    }
+  }, [edit]);
+
+  function handleChange(word: string) {
+    const trimmed = word.trim();
+    if (trimmed.length <= 32) {
+      setKeyword(word.trim());
+    }
+  }
+
+  function handleAdd(word: string) {
+    try {
+      addCb(word);
+      setKeyword('');
+      setEdit(false);
+    } catch (error: any) {
+      eventEmitter.emit('error', {
+        name: 'Add Keyword',
+        message: error?.message ?? 'Cannot add Keyword',
+      });
+      return false;
+    }
+  }
+
+  return (
+    <>
+      {!edit ? (
+        <button
+          onClick={() => setEdit(true)}
+          className="flex flex-row rounded-full bg-background border border-dark-grey shadow-3xl py-1 items-center justify-center px-[10px] max-w-fit h-fit text-grey hover:text-white hover:border-grey transition-all"
+          style={{ gap: '10px' }}
+        >
+          <PlusCircleFilled
+            width={'18px'}
+            height={'18px'}
+            className="text-success"
+          />{' '}
+          <span className="flex flex-row max-w-fit h-fit font-semibold items-center">
+            add
+          </span>
+        </button>
+      ) : (
+        <div className="flex flex-row max-w-[330px]" style={{ gap: '10px' }}>
+          <input
+            ref={inputRef}
+            className={`flex flex-row rounded-full bg-background border border-dark-grey shadow-3xl py-[2px] items-center justify-center px-[10px] w-fit w-fit text-white transition-all`}
```

Comment:
Change px to rem

Reply:
apply throughout PR

Reply:
[2b04cdf](https://github.com/ar-io/arns-react/pull/620/commits/2b04cdff991e2f1b828fef7102a1b73d938e65ee)